src/Microsoft.AspNetCore.Sockets.Client/Connection.cs

```diff
@@ -21,6 +20,7 @@ public class Connection: IConnection
         private IChannelConnection<Message> _transportChannel;
         private ITransport _transport;
         private Task _receiveLoopTask;
+        private Task _startTask = Task.CompletedTask;
```

Comment:
Should _startTask and _connectionState be volatile? I would expect in the time it takes an HTTP request to complete in GetConnectUrl, the disposing thread will see the Task, but better to be safe.